qutebrowser/browser/qutescheme.py

```diff
@@ -158,6 +161,59 @@ def qute_bookmarks(_url):
     return 'text/html', html
 
 
+@add_handler('history')
+def qute_history(url):
+    """Handler for qute:history. Display history."""
+    # Get current date from query parameter, if not given choose today.
+    curr_date = datetime.date.today()
+    try:
+        query_date = QUrlQuery(url).queryItemValue("date")
+        if query_date:
+            curr_date = datetime.datetime.strptime(query_date, "%Y-%m-%d")
+            curr_date = curr_date.date()
+    except ValueError:
+        log.misc.debug("Invalid date passed to qute:history: " + query_date)
+
+    one_day = datetime.timedelta(days=1)
+    next_date = curr_date + one_day
+    prev_date = curr_date - one_day
+
+    def history_iter():
+        for item in objreg.get('web-history').history_dict.values():
```

Comment:
I think it does not matter much, because this loop is over the whole history in any case, there would still be a generator -> `list` conversion and calling `reversed` on the list does not create another copy. The problem is that filtering for just one day only means that the iteration will `continue` most of the time, but python's performance is terrible on (mostly) empty iterations. You'd need an indexed history to fix this.